entry.sh

```diff
@@ -79,6 +79,19 @@ if [ -w /etc/authorized_keys ]; then
     done
 fi
 
+# Add groups if SSH_GROUPS=group:gid set
+if [ -n "${SSH_GROUPS}" ]; then
+    USERS=$(echo $SSH_USERS | tr "," "\n")
+    GROUPZ=$(echo $SSH_GROUPS | tr "," "\n")
```

Comment:
It took me quite a long time to discover that the natural name for this variable, GROUPS, is a reserved keyword in bash, and assigning to it silently kills the script!